### DISCUSSION TITLE
Testing Local Data Storage (MSTG-STORAGE-1 and MSTG-STORAGE-2) - Update Keychain attributes

### DISCUSSION THREAD

**Discussion by rsenet** [OWASP/mastg#2353] [Ideas]
In https://mas.owasp.org/MASTG/iOS/0x06d-Testing-Data-Storage/#the-keychain, it could be interesting to remove **kSecAttrAccessibleAlways** and **kSecAttrAccessibleAlwaysThisDeviceOnly** are [deprecated](https://developer.apple.com/documentation/security/keychain_services/keychain_items/item_attribute_keys_and_values#1679100).

Moreover, it could be interesting to add **kSecAccessControlWatch** in **AccessControlFlags**
Apple documentation: https://developer.apple.com/documentation/security/secaccesscontrolcreateflags/ksecaccesscontrolwatch

**Comment by cpholguera** [comment]
@rsenet first of all let me apologize for the very late answer. We've been working on the new MAS profiles and MASTG Atomic Tests. Have you seen the news?

https://mas.owasp.org/news/

I'd love to see you participate, you're always giving us great feedback and awesome ideas. Here's the proposal for MASTG tests:

https://docs.google.com/spreadsheets/d/1Go5GpVvKJqTDxGbSLBPZb1hmYi5lXRc1D1AfrTTkUkY/edit?usp=sharing

Feel free to enter your feedback in any cell. Once the proposal is finalized and reviewed we'll start implementing all those tests. I really hope we can count on you for that. Thanks a lot!

**Comment by cpholguera** [comment]
Regarding this topic: absolutely, yes! We need to update that information. Thanks for keeping an eye on that!
